### DISCUSSION TITLE
celery.Asyncresult memory leak

### DISCUSSION THREAD

**Discussion by TXYH1** [celery/celery#8480] [Q&A]
I use python27 + celery4.4.7, provide an http service, execute submitted tasks through subprocesses, and provide an interface to obtain historical task status in the http service

I get task information through `celery.AsyncResult(task_id)` and access task.state, task.result. But this AsyncResult will never be recycled, no matter how many times gc.collect() is called, this object will always exist in gc.garbage, until the container memory is exhausted and OOM

objgraph.show_backrefs look like
<img width="424" alt="image" src="https://github.com/celery/celery/assets/30724518/9fcdeb00-42dc-4fda-b67c-87033d8fa6be">

How to deal with this situation is better

Thank you

**Comment by auvipy** [comment]
Help us to reproduce this with celery 5.3.1 / main branch please

**Comment by TXYH1** [comment]
But python2 can't use a version higher than celery4.4.7, I can't upgrade python, so I want to see if there is any solution for celery4 under py27

**Comment by auvipy** [comment]
we don't have python 2 support.

**Comment by TXYH1** [comment]
Okay, I understand that maintaining on older versions is not necessary, but your excuse is not very good.

<img width="852" alt="image" src="https://github.com/celery/celery/assets/30724518/3b008219-9748-4db3-bbce-f3a1447bfbcf">
